content/en/blog/_posts/2021-03-03-using-finalizers-to-control-deletion.md

```diff
@@ -0,0 +1,269 @@
+---
+layout: blog
+title: 'Using Finalizers to Control Deletion'
+date: 2021-03-03
+slug: using-finalizers-to-control-deletion
+---
+
+**Authors:** Aaron Alpar (Kasten)
+
+Deleting objects in Kubernetes can be challenging. You may think you’ve deleted something, only to find it still persists. While issuing a `kubectl delete` command and hoping for the best might work for day-to-day operations, understanding how Kubernetes `delete` commands operate will help you understand why some objects linger after deletion. 
+
+In this post, I’ll look at:
+
+- What properties of a resource govern deletion
+- How finalizers and owner references impact object deletion
+- How the propagation policy can be used to change the order of deletions
+- How deletion works, with examples
+
+For simplicity, all examples will use configmaps and basic shell commands to demonstrate the process. We’ll explore how the commands work and discuss repercussions and results from using them in practice.
+
+## The basic `delete`
+
+Kubernetes has several different commands you can use that allow you to create, read, update, and delete objects. For the purpose of this blog post, we’ll focus on four `kubectl` commands: `create`, `get`, `patch`, and `delete`. 
+
+Here are examples of the basic `kubectl delete` command:
+
+```
+$ kubectl create configmap mymap
+configmap/mymap created
+```
+
+```
+$ kubectl get configmap/mymap
+NAME    DATA   AGE
+mymap   0      12s
+```
+
+```
+$ kubectl delete configmap/mymap
+configmap "mymap" deleted
+```
+
+```
+$ kubectl get configmap/mymap
+Error from server (NotFound): configmaps "mymap" not found
+```
+
+Shell commands preceded by `$` are followed by their output. You can see that we begin with a `kubectl create configmap mymap`, which will create the empty configmap `mymap`. Next, we need to `get` the configmap to prove it exists. We can then delete that configmap. Attempting to `get` it again produces an HTTP 404 error, which means the configmap is not found.
+
+The state diagram for the basic `delete` command is very simple:
+
+![State diagram for delete](/images/blog/2021-03-03-use-finalizers-to-control-deletion/state-diagram-delete.png)

```

Comment:
This doesn't render - see https://deploy-preview-26997--kubernetes-io-master-staging.netlify.app/blog/2021/03/03/using-finalizers-to-control-deletion/

Reply:
interesting ... I'll reproduce and fix here.